src/autorest.bicep/src/type-generator.ts

```diff
@@ -294,13 +294,37 @@ export function generateTypes(host: Host, definition: ProviderDefinition) {
     return factory.lookupBuiltInType(BuiltInTypeKind.Any);
   }
 
+  function getMutabilityFlags(property: Property | undefined) {
+    const mutability = property?.extensions?.["x-ms-mutability"] as string[];
```

Comment:
>x-ms-mutability [](http://example.com/codeflow?start=47&length=15)

I think you also need to handle x-ms-secret: true and format: password cases. I sent you an IM about how we're dealing with that in an internal repo that is partially applicable.

If something is considered secret, we should assume write only and it should take precedence over whatever x-ms-mutability settings in the swagger.

Reply:
I've raised https://github.com/Azure/bicep-types-az/issues/437 to track this separately as this PR is just to fix a specific scenario. I feel like the handling of `x-ms-secret` in types needs a bit more discussion.

Reply:
Sounds good.